src/components/Preferences/LocalIdentities.js

```diff
@@ -22,88 +23,129 @@ import Import from './Import'
 
 const iOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream
 
-const LocalIdentities = ({
-  dao,
-  localIdentities,
-  onClearAll,
-  onImport,
-  onModify,
-  onModifyEvent,
-}) => {
-  // transform localIdentities from object into array and attach address to each entry
-  const identities = Object.entries(localIdentities).map(
-    ([address, identity]) => ({
-      ...identity,
-      address,
-    })
-  )
-
-  if (!identities.length) {
-    return <EmptyLocalIdentities onImport={onImport} />
-  }
+const LocalIdentities = React.memo(
+  ({ dao, localIdentities, onClearAll, onImport, onModify, onModifyEvent }) => {
+    // transform localIdentities from object into array and attach address to each entry
+    const identities = Object.entries(localIdentities).map(
+      ([address, identity]) => ({
+        ...identity,
+        address,
+      })
+    )
 
-  const { identityEvents$ } = React.useContext(IdentityContext)
-  const { showLocalIdentityModal } = React.useContext(LocalIdentityModalContext)
-  const updateLabel = address => async () => {
-    try {
-      await showLocalIdentityModal(address)
-      // preferences get all
-      onModifyEvent()
-      // for iframe apps
-      identityEvents$.next({ type: identityEventTypes.MODIFY, address })
-    } catch (e) {
-      /* nothing was updated */
+    if (!identities.length) {
+      return <EmptyLocalIdentities onImport={onImport} />
     }
-  }
-  const downloadHref = window.URL.createObjectURL(
-    new Blob([JSON.stringify(identities)], { type: 'text/json' })
-  )
-  // standard: https://en.wikipedia.org/wiki/ISO_8601
-  const today = format(Date.now(), 'yyyy-MM-dd')
-
-  return (
-    <React.Fragment>
-      <Headers>
-        <div>Custom label</div>
-        <div>Address</div>
-      </Headers>
-      <List>
-        {identities.map(({ address, name }) => (
-          <Item key={address}>
-            <Label>{name}</Label>
-            <div>
-              <IdentityBadge
-                entity={address}
-                popoverAction={{
-                  label: 'Edit custom label',
-                  onClick: updateLabel(address),
-                }}
-                popoverTitle={<LocalIdentityPopoverTitle label={name} />}
+
+    const { identityEvents$ } = React.useContext(IdentityContext)
+    const { showLocalIdentityModal } = React.useContext(
+      LocalIdentityModalContext
+    )
+    const updateLabel = address => async () => {
+      try {
+        await showLocalIdentityModal(address)
+        // preferences get all
+        onModifyEvent()
+        // for iframe apps
+        identityEvents$.next({ type: identityEventTypes.MODIFY, address })
+      } catch (e) {
+        /* nothing was updated */
+      }
+    }
+    // standard: https://en.wikipedia.org/wiki/ISO_8601
+    const today = format(Date.now(), 'yyyy-MM-dd')
+    const [addressSelected, setAddressSelected] = React.useState(
```

Comment:
What do you think about moving this up to where the `useContext()`s are? I think it's better if we group most of the hooks together.

Also, what do you think about renaming it to `addressesSelected` (since there's more than one address being selected)?